frontend/src/routes/ScoreboardsPage.js

```diff
@@ -171,104 +46,55 @@ export default function ScoreboardsPage() {
         },
       }}
     >
-      <Typography sx={{ m: 4 }} variant="h4">
-        Best Models
-      </Typography>
-      {adminMode ? <AdminPanel refreshFunc={refresh} /> : null}
-      <Card sx={{ maxWidth: '90vw', m: 4 }}>
-        <DataTable
-          columns={fittingColumns}
-          rows={fittingRows}
-          highlightedRows={highlightedRows}
+      {adminMode ? <AdminPanel refreshFunc={refreshFuncs[activeTab]} /> : null}
+      <Tabs value={activeTab} onChange={handleChange}>
+        <Tab label="Best Models" />
+        <Tab label="Best Molecules" />
+      </Tabs>
+      <TabPanel value={activeTab} index={0}>
+        <BestModels
+          passRefreshFunc={(refreshFunc) => {
+            refreshFuncs[0] = refreshFunc
+          }}
+          datasets={datasets}
         />
-      </Card>
-    </Box>
-  )
-}
-
-/**
- * a hidden panel with extra features which shows up when  in adminMode
- * @param refreshFunc gets called when the button inside this panel "Delete All!" is clicked
- * @returns {JSX.Element} a box containing a button that triggers the given function when clicked or nothing when adminMode == false
- */
-function AdminPanel({ refreshFunc }) {
-  return (
-    <Box>
-      <Button
-        variant="contained"
-        size="large"
-        sx={{ m: 1 }}
-        onClick={() => {
-          api.deleteScoreboardFittings().then(() => {
-            refreshFunc()
-          })
-        }}
-        endIcon={<DeleteSweepIcon />}
-      >
-        Delete All!
-      </Button>
+      </TabPanel>
+      <TabPanel value={activeTab} index={1}>
+        <BestMolecules
+          passRefreshFunc={(refreshFunc) => {
+            refreshFuncs[1] = refreshFunc
```

Comment:
This does not work. Site crashes when "Delete All" is clicked. No matter the scoreboard tab.

Also I personally don't like the way refresh functions are handled here. Is there a different way to do this?

Reply:
Ah, yes, forgot that existed. Will implement.
Also will brainstorm alternatives to refreshFuncs.

Reply:
I moved the admin panel so each tab has its own instance and can pass its deletion trigger down naturally.